Exiled.Events/Patches/Events/Player/EarningAchievement.cs

```diff
@@ -0,0 +1,78 @@
+// -----------------------------------------------------------------------
+// <copyright file="EarningAchievement.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Events.Player
+{
+    using System.Collections.Generic;
+    using System.Reflection.Emit;
+
+    using Achievements;
+
+    using API.Features;
+    using API.Features.Pools;
+    using Exiled.Events.EventArgs.Player;
+
+    using HarmonyLib;
+
+    using Mirror;
+
+    using static HarmonyLib.AccessTools;
+
+    /// <summary>
+    ///     Patch the <see cref="AchievementHandlerBase.ServerAchieve"/>.
+    ///     Adds the <see cref="Handlers.Player.EarningAchievement"/> event.
+    /// </summary>
+    [HarmonyPatch(typeof(AchievementHandlerBase), nameof(AchievementHandlerBase.ServerAchieve))]
```

Comment:
Missing the new EventPatch argument (look other event)

Reply:
done